examples/04_fuzzy_joining.py

```diff
@@ -362,51 +362,34 @@
 # .............................
 
 y = df["Happiness score"]
-#######################################################################
-# We gather the auxilliary tables into a
-# list of (tables, keys) for the `tables` parameter.
-# An instance of the transformer with the necessary information is:
-from skrub import Joiner
-
-joiner = Joiner(
-    tables=[
-        (gdppc, "Country Name"),
-        (life_exp, "Country Name"),
-        (legal_rights, "Country Name"),
-    ],
-    main_key="Country",
-)
-
-#################################################################
-# Fitting and transforming into the final table
-# .............................................
-# To get our final joined table we will fit and transform the main table (df)
-# with our create instance of the |joiner|:
-df_final = joiner.fit_transform(df)
-
-df_final.head(10)
-
-##########################################################################
-# And that's it! As previously, we now have a big table
-# ready for machine learning.
-# Let's create our machine learning pipeline:
-from sklearn.compose import make_column_transformer
-from sklearn.pipeline import make_pipeline
-
-# We include only the columns that will be pertinent for our regression:
-encoder = make_column_transformer(
-    (
-        "passthrough",
-        [
-            "GDP per capita (current US$)",
-            "Life expectancy at birth, total (years)",
-            "Strength of legal rights index (0=weak to 12=strong)",
-        ],
-    ),
-    remainder="drop",
+df = df.drop("Happiness score", axis=1)
+
+from skrub import Joiner, SelectCols, DropCols
+from sklearn.pipeline import Pipeline
+
+steps = [
+    ("gdppc", Joiner((gdppc, "Country Name"), "Country")),
+    ("gdppc-drop", DropCols(["Country Name"])),
+    ("life_exp", Joiner((life_exp, "Country Name"), "Country")),
+    ("life_exp_drop", DropCols(["Country Name"])),
+    ("legal_rights", Joiner((legal_rights, "Country Name"), "Country")),
+]
+
+# At the end of our pipeline, we project on the relevant attributes and finally
+# fit our regressor
+
+select = SelectCols(
+    [
+        "GDP per capita (current US$)",
+        "Life expectancy at birth, total (years)",
+        "Strength of legal rights index (0=weak to 12=strong)",
+    ]
 )
-
-pipeline = make_pipeline(joiner, encoder, HistGradientBoostingRegressor())
+steps = steps + [
+    ("select", select),
+    ("regressor", HistGradientBoostingRegressor()),
+]
```

Comment:
Why not build the "steps" list in one go.

Also, people typically find "make_pipeline" more readable. I would argue for using it here.

Reply:
I felt like putting the `SelectCols` in a separate statement because it adds several lines and levels of indentation due to the long column names and black formatting, but I can put it before and make the whole pipeline in one statement.

I also prefer "make_pipeline" so I'll change it, the drawback is as we have several joiners we have to refer to them with "Joiner-1", "Joiner-2" in the gridsearch

Reply:
> I also prefer "make_pipeline" so I'll change it, the drawback is as we have several joiners we have to refer to them with "Joiner-1", "Joiner-2" in the gridsearch

actually that's not so bad even if it's less obvious where the names came from